functions/Move-DbaDbFile.ps1

```diff
@@ -0,0 +1,366 @@
+
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database, Move, File
+        Author: Cláudio Silva (@claudioessilva), claudioeesilva.eu
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Move-DbaDbFile
+
+    .EXAMPLE
+        PS C:\> Move-DbaDbFile -SqlInstance sql2017 -Database dbatools -FileType Data -FileDestination "D:\DATA2"
+
+        Copy all data files of dbatools database on sql2017 instance to the "D:\DATA2" path.
+        Before it puts database offline and after copy each file will update database metadata and it ends by set the database back online
+
+    .EXAMPLE
+        PS C:\> $fileToMove=@{
+            'dbatools'='D:\DATA3'
+            'dbatools_log'='D:\LOG2'
+        }
+        PS C:\> Move-DbaDbFile -SqlInstance sql2019 -Database dbatools -FileToMove $fileToMove
+
+        Declares a hashtable that says for each logical file the new path.
+        Copy each dbatools database file referenced on the hashtable on the sql2019 instance from the current location to the new mentioned location (D:\DATA3 and D:\LOG2 paths).
+        Before it puts database offline and after copy each file will update database metadata and it ends by set the database back online
+
+    .EXAMPLE
+        PS C:\> Move-DbaDbFile -SqlInstance sql2017 -Database dbatools -FileStructureOnly
+
+        Shows the current database file structure (without filenames).
+
+    #>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(Mandatory)]
+        [string]$Database,
+        [parameter(ParameterSetName = "All")]
+        [ValidateSet('Data', 'Log', 'Both')]
+        [string]$FileType,
+        [parameter(ParameterSetName = "All")]
+        [string]$FileDestination,
+        [parameter(ParameterSetName = "Detailed")]
+        [hashtable]$FileToMove,
+        [parameter(ParameterSetName = "All")]
+        [parameter(ParameterSetName = "Detailed")]
+        [switch]$DeleteAfterMove,
+        [parameter(ParameterSetName = "FileStructure")]
+        [switch]$FileStructureOnly,
+        [switch]$EnableException
+    )
+
+    begin {
+        if ((Test-Bound -ParameterName FileType) -and (-not(Test-Bound -ParameterName FileDestination))) {
+            Stop-Function -Category InvalidArgument -Message "You need to specify the -FileDestination parmeter. Quitting."
+        }
+    }
+
+    process {
+        if (Test-FunctionInterrupt) { return }
+
+        if (!$FileType -and !$FileToMove -and !$FileStructureOnly ) {
```

Comment:
Please expand `!` to use `-not` for readability.

Reply:
Done